src/Uno.UI.Runtime.Skia.Gtk/GtkCorePointerInputSource.cs

```diff
@@ -0,0 +1,591 @@
+		//		   That's the purpose of the UnoEventBox.
+		GtkHost.EventBox.EnterNotifyEvent += OnEnterEvent;
+		GtkHost.EventBox.LeaveNotifyEvent += OnLeaveEvent;
+		GtkHost.EventBox.ButtonPressEvent += OnButtonPressEvent;
+		GtkHost.EventBox.ButtonReleaseEvent += OnButtonReleaseEvent;
+		GtkHost.EventBox.MotionNotifyEvent += OnMotionEvent;
+		GtkHost.EventBox.ScrollEvent += OnScrollEvent;
+		GtkHost.EventBox.Touched += OnTouchedEvent; //Note: we don't use the TouchEvent for the reason explained in the UnoEventBox!
+		GtkHost.EventBox.ProximityInEvent += OnProximityInEvent;
+		GtkHost.EventBox.ProximityOutEvent += OnProximityOutEvent;
+	}
+
+	/// <inheritdoc />
+	[NotImplemented]
+	public bool HasCapture => false;
+
+	/// <inheritdoc />
+	[NotImplemented]
+	public Windows.Foundation.Point PointerPosition => default;
+
+	/// <inheritdoc />
+	public CoreCursor PointerCursor
+	{
+		get => GtkHost.Window.Window.Cursor.ToCoreCursor();
+		set => GtkHost.Window.Window.Cursor = value.ToCursor();
+	}
+
+	/// <inheritdoc />
+	public void SetPointerCapture()
+	{
+		if (_lastUsedDevice is not null)
+		{
+			Gtk.Device.GrabAdd(GtkHost.Window, _lastUsedDevice, block_others: false);
+		}
+		else if(this.Log().IsEnabled(LogLevel.Error))
+		{
+			this.Log().Error("No native device to capture.");
+		}
+	}
+
+	/// <inheritdoc />
+	public void SetPointerCapture(PointerIdentifier pointer)
+	{
+		if (_knownDevices.TryGetValue(pointer, out var entry))
+		{
+			Gtk.Device.GrabAdd(GtkHost.Window, entry.dev, block_others: false);
+		}
+		else if (this.Log().IsEnabled(LogLevel.Error))
+		{
+			this.Log().Error($"Unknown device id {pointer}, unable to natively capture it.");
+		}
+	}
+
+	/// <inheritdoc />
+	public void ReleasePointerCapture()
+	{
+		if (_lastUsedDevice is not null)
+		{
+			Gtk.Device.GrabAdd(GtkHost.Window, _lastUsedDevice, block_others: false);
+		}
+		else if (this.Log().IsEnabled(LogLevel.Error))
+		{
+			this.Log().Error("No native device to release.");
+		}
+	}
+
+	/// <inheritdoc />
+	public void ReleasePointerCapture(PointerIdentifier pointer)
+	{
+		if (_knownDevices.TryGetValue(pointer, out var entry))
+		{
+			Gtk.Device.GrabRemove(GtkHost.Window, entry.dev);
+		}
+		else if (this.Log().IsEnabled(LogLevel.Error))
+		{
+			this.Log().Error($"Unknown device id {pointer}, unable to natively capture it.");
+		}
+	}
+
+	#region Native event handlers
+	private void OnEnterEvent(object o, EnterNotifyEventArgs args)
+	{
+		try
+		{
+			if (AsPointerArgs(args.Event) is { } ptArgs)
+			{
+				RaisePointerEntered(ptArgs);
+			}
+		}
+		catch (Exception e)
+		{
+			this.Log().Error("Failed to raise PointerEntered", e);
+		}
+	}
+
+	private void OnLeaveEvent(object o, LeaveNotifyEventArgs args)
+	{
+		try
+		{
+			// The Ungrab mode event is triggered after click even when the pointer does not leave the window.
+			// This may need to be removed when we implement native pointer capture support properly.
```

Comment:
```suggestion
			// The Ungrab mode event is triggered after clicking even when the pointer does not leave the window.
			// This may need to be removed when we implement native pointer capture support properly.
```